src/lib/services/transfer-service/TransferService.ts

```diff
@@ -169,6 +203,80 @@ export class TransferService {
         }
     }
 
+    private async putThumbnailImageToShell(aas: AssetAdministrationShell, apikey?: string): Promise<TransferResult> {
+        try {
+            const aasThumbnail = await this.sourceAasRepositoryClient.getThumbnailFromShell(aas.id);
+            const fileName = ['thumbnail', generateRandomNumber()].join('');
+            await this.targetAasRepositoryClient.putThumbnailToShell(aas.id, aasThumbnail, fileName, {
+                headers: {
+                    Apikey: apikey,
+                },
+            });
+            return { success: true, operationKind: 'AasRepository', resourceId: 'Thumbnail transfer.', error: '' };
+        } catch (e) {
+            return {
+                success: false,
+                operationKind: 'AasRepository',
+                resourceId: 'Thumbnail transfer.',
+                error: e.toString(),
+            };
+        }
+    }
+
+    private async processAttachments(submodelId: string, attachmentDetails: AttachmentDetails[], apikey?: string) {
+        const promises = [];
+
+        for (const attachmentDetail of attachmentDetails) {
+            try {
+                attachmentDetail.file = await this.sourceSubmodelRepositoryClient.getAttachmentFromSubmodelElement(
+                    submodelId,
+                    attachmentDetail.idShortPath,
+                );
+                this.processFileExtension(attachmentDetail);
+                promises.push(await this.putAttachmentToSubmodelElement(submodelId, attachmentDetail, apikey));
+            } catch (e) {
+                promises.push({
+                    success: false,
+                    operationKind: 'SubmodelRepository',
+                    resourceId: [
+                        'File transfer: ',
```

Comment:
maybe rename the operation kind to File Transfer and the resourceId to some combination of submodel and shortId